### DISCUSSION TITLE
Linux Cluster

### DISCUSSION THREAD

**Discussion by jfpe37** [firemodels/fds#12746] [General]
Hi
We are looking into the possibility of moving to running fds simolations on a Linux server.
Is there a guide that explains how to do this?
based on your experience are there any recommendations?
1. Which Linux to install?
2. Which version (for example 20.04 Ubuntu server)
3. Recommended amount of processors and memory for working with large models (around 10 million cells)
Thanks in advance

**Comment by rmcdermo** [comment]
Others are more expert than I, but since we just procured a new machine, I can give you some ideas to consider.

1. Enterprise Linux (EL) and its derivatives seem to be recommended for HPC systems, though I have heard of Ubuntu being used as well.  We had planned to use Rocky 9, but now are being forced to use Red Hat due to FIPS (encryption) certification.
2. We were planning on latest (Rocky 9.3), but if you are more comfortable with Ubuntu, then I do not have an argument against it.  My understanding is that EL systems tend to be more stable.
3. FDS scales reasonably well up to about 16^3 (4096) cells/mesh.  More realistically you could think in terms of 32^3 cells per mesh, which would require 305 cores (1 mesh/core) for a 10 million cell calculation.  The machine we are getting has 64 cores/node and 8 GB RAM/core.  I would focus on the RAM / core---you should have at least 4 GB/core.  This is the memory that each mesh needs to operate.  So, if you had the same nodes as our system, you could do your 10 million cell job with 5 nodes.

Another decision you will have to make is Intel versus AMD.  Lots of opinions here.  We have stayed with Intel since we use the OneAPI compilers.  However, there is a lot of discussion about ARM chips in the future, but I suspect that is 5 years away for us.

**Reply by JakeOShannessy** [reply]
I'd concur with points 1 and 2. We use RHEL based distributions (mainly Rocky but likely Alma Linux soon). Partially for the stability but mainly for the network effects. Many things in the HPC space are well supported on these distributions.

I expect you could build and run FDS on Ubuntu equally but we've had little issue with RHEL based distributions.

**Comment by jfpe37** [comment]
Thanks
Are there any other recommendations?
I would appreciate your help.
Mor

**Comment by drjfloyd** [comment]
We use RHEL 9.3 (Plow) for our system and it works fine.

Are you looking for just a single machine or a rack system with multiple server blades in it?  For a rack system you are going to want a fast, high bandwidth network (we use InfiniBand).  

My experience is FDS scales well for larger meshes than Randy indicated.   You could do a 10 million cell case on a single 64-core node (~50^3 mesh). It will obviously not run as fast as it would if you had more meshes with fewer cells per mesh.

Disk storage is another concern.  You want fast disks and a lot of storage. We have 20 Tb of strorage and have had single jobs that have consumed a third of it.

Another thing to consider, especially with a rack system, is cooling. Each of our nodes disipates about 1 kW of heat when running at 100 % load.

**Comment by rmcdermo** [comment]
Yes, the scaling depends on the system, and I was referring to national class HPC system where these strong scaling tests were performed.  But this brings up a good point that we have not mentioned yet, which is you need to consider the Infiniband (IB) fabric.  Do not try to do HPC with ethernet.  I think by now that Mellanox is the standard.  We spec'd HDR100 and a Mellanox Quantum (200 Gb/s) IB switch.
